### PR TITLE
Basic implementation of agent

### PR DESCRIPTION
**Pull request by fmarek-kindred**

### CODE REVIEW

File: packages/talos_agent/src/agent.rs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+    /// Starts listener task which reads Talos decisions from the topic
+    pub fn start(&self, int_type: &TalosIntegrationType) -> Result<(), String> {
+        let agent_id = self.config.agent_id.clone();
+        let config = self.kafka_config.clone().expect("Kafka configuration is required");
+        let in_flight = Arc::clone(&self.in_flight);
+
+        let it = int_type.clone();
+        tokio::spawn(async move {
+            let consumer: Box<ConsumerType> = match it {
```
Comment (gk-kindred) [inline packages/talos_agent/src/agent.rs:66]: Would be nice in future iterations to be able to pass the Consumer based on `trait` and therefore I can pass the actual consumer or any mock to run my tests. 

Currently I can pass only the real consumer or a mock which will return `None` on receive message. 

Not needed in this PR, but would be great to have this refactored later.

File: packages/talos_agent/src/api.rs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    pub fn build(&self) -> Result<Box<TalosAgentType>, String> {
+        let publisher: Box<PublisherType> = match self.integration_type {
+            Kafka => {
+                let config = &self.kafka_config.clone().expect("Kafka configuration is required");
+                let kafka_publisher = KafkaPublisher::new(self.config.agent_id.clone(), config);
+                Box::new(kafka_publisher)
+            }
+            _ => Box::new(MockPublisher {}),
```
Comment (gk-kindred) [inline packages/talos_agent/src/api.rs:111]: Same comment as the `consumer`. A particular implementation for Mock is tightly coupled here. We should make it flexible to pass different Mocks to test different scenarios.

File: packages/talos_agent/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        if d < min {
+            min = d;
+        }
+        if d > max {
+            max = d;
+        }
+        debug!("Transaction has been certified. Details: {:?}", rsp);
+    }
+
+    metrics.sort();
```
Comment (gk-kindred) [inline packages/talos_agent/src/main.rs:134]: nice one with the metrics :)

**Review by gk-kindred — APPROVED** [review]

File: packages/talos_agent/src/api.rs (hunk 2 above)
Comment (fmarek-kindred) [reply packages/talos_agent/src/api.rs:111]: I agree, currently it is just a placeholder really. I will change this part.

**Pull request merged**